### DISCUSSION TITLE
Split assertion conjuncts for struct equality

### DISCUSSION THREAD

**Discussion by jonhnet** [verus-lang/verus#447] [General]
Here's an assertion where it would be awesome to get a "split assertion failure" helper note about which of the fields didn't match up:

```
error: assertion failed
   --> src/host_impl_v.rs:220:21
    |
220 |               assert( constants@ == AbstractConstants{
    |  _____________________^
221 | |                 root_identity: abstract_end_points@.unwrap()[0],
222 | |                 host_ids: abstract_end_points@.unwrap(),
223 | |                 params: AbstractParameters::static_params(),
224 | |                 me: me@ } );
    | |_________________________^ assertion failed
```

As it is, I have to copy/paste each field into a separate equality assertion to break down the problem. That's exactly what "split assertion failures" are for.

(Side note: "split assertion failure" is a strange label; I don't know what it means. Maybe we can come up with a better description to help new people read the feedback.)

**Comment by tjhance** [comment]
+1 this would be awesome
